### PR TITLE
k9s: 0.25.18 -> 0.25.21

### PR DESCRIPTION
**Pull request by bryanasdev000**
###### Description of changes

- bump version

- add missing ldflags

- add missing go tags

- add install check

- add shell completion

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/cluster/k9s/default.nix [hunk 1]
```diff
@@ -1,28 +1,43 @@
...
+  vendorSha256 = "sha256-19P1GK4BoJBav7x6NLVFl256++DuxN2zGRWEY6i8POI=";
 
   preCheck = "export HOME=$(mktemp -d)";
 
   # TODO investigate why some config tests are failing
   doCheck = !(stdenv.isDarwin && stdenv.isAarch64);
+  doInstallCheck = true;
+  installCheckPhase = ''
+    $out/bin/k9s version | grep ${version} > /dev/null
+  '';
```
Comment (thiagokokada) [inline pkgs/applications/networking/cluster/k9s/default.nix:32]: What about using `testVersion` instead 🤔 ?

Comment (bryanasdev000) [reply pkgs/applications/networking/cluster/k9s/default.nix]: https://media1.giphy.com/media/xT5LMzIK1AdZJ4cYW4/giphy.gif?cid=ecf05e47lwuwjsq8em2v4jae43whqiedfxg4msyi67tyyzrr&rid=giphy.gif&ct=g

Done.

**Comment by thiagokokada** [comment]
Tests are failing in all platforms.,

**Comment by bryanasdev000** [comment]
> Tests are failing in all platforms.

Gonna check it, seems that even with a version string in stdout, the test fails.

**Comment by bryanasdev000** [comment]
> > > Tests are failing in all platforms.
> > 
> > 
> > Gonna check it, seems that even with a version string in stdout, the test fails.
> 
> From the error logs, I think maybe you will need to "create" a `HOME` directory here.

Thanks for the feedback @thiagokokada, I gonna update the package again and test later.

Will leave as a draft for now.

**Comment by thiagokokada** [comment]
> > > > Tests are failing in all platforms.
> > > 
> > > 
> > > Gonna check it, seems that even with a version string in stdout, the test fails.
> > 
> > 
> > From the error logs, I think maybe you will need to "create" a `HOME` directory here.
> 
> Thanks for the feedback @thiagokokada, I gonna update the package again and test later.
> 
> Will leave as a draft for now.

If this gets too complicated you can go back to `installCheck`.

However I think it should work if you set `command = "HOME=$(mktemp -d) k9s version`. If it doesn't because `mktemp` doesn't exist, setting `HOME=. k9s version` should also work.

**Comment by bryanasdev000** [comment]
> > > > > Tests are failing in all platforms.
> > > > 
> > > > 
> > > > Gonna check it, seems that even with a version string in stdout, the test fails.
> > > 
> > > 
> > > From the error logs, I think maybe you will need to "create" a `HOME` directory here.
> > 
> > 
> > Thanks for the feedback @thiagokokada, I gonna update the package again and test later.
> > Will leave as a draft for now.
> 
> If this gets too complicated you can go back to `installCheck`.
> 
> However I think it should work if you set `command = "HOME=$(mktemp -d) k9s version`. If it doesn't because `mktemp` doesn't exist, setting `HOME=. k9s version` should also work.

Thanks, [@thiagokokada](https://github.com/thiagokokada), actually I was doing two things and the other required some attention, also I think that tests are broken upstream, so I will keep this as a draft for now.

**Comment by bryanasdev000** [comment]
For ref: https://github.com/derailed/k9s/issues/1626

**Comment by bryanasdev000** [comment]
@thiagokokada seems ready to ship, could you please take a look?

Thank you!

**Comment by bryanasdev000** [comment]
Sorry to ping again @thiagokokada, but can you please take a look again?

Thanks in advanced.

File: pkgs/applications/networking/cluster/k9s/default.nix [hunk 2]
```diff
@@ -1,28 +1,45 @@
...
-  vendorSha256 = "sha256-mMob7M9RQlqaVK0DgHpaAK9d1btzfQetnliUqFTvjJQ=";
+  tags = [ "netgo" ];
 
-  preCheck = "export HOME=$(mktemp -d)";
+  vendorSha256 = "sha256-wL8Unht/ZRAGDuC/U4SFV5PdExy78F4DMyM8+7CMtOY=";
 
   # TODO investigate why some config tests are failing
   doCheck = !(stdenv.isDarwin && stdenv.isAarch64);
+  # Required to workaround install check error:
+  preCheck = "export HOME=$(mktemp -d)";
```
Comment (thiagokokada) [inline pkgs/applications/networking/cluster/k9s/default.nix:28]: Do you still need this? Doesn't seem like you're doing any kinda of install check.

**Review by thiagokokada — APPROVED** [review]
Code-wise, LGTM.

**Comment by bryanasdev000** [comment]
@thiagokokada looking in tests logs for darwin and arm64, seems that preCheck is needed.

**Review by thiagokokada — APPROVED** [review]

**Pull request merged**